[Bug]: Test failing

### Brief problem description

Expected to pass all 4 test
Result: FAILED tests/test_items.py::test_get_item_1 - assert 404 == 200

### Steps to Reproduce

1: In the terminal, run ```uv run poe test```

### Expected Result

0 failed, 4 passed

### Actual Result

====================================================================== FAILURES =======================================================================
___________________________________________________________________ test_get_item_1 ___________________________________________________________________

client = <starlette.testclient.TestClient object at 0x7f7bbb593d90>

    def test_get_item_1(client: TestClient):
>       get_task_by_id(
            client=client,
            task_id="lab-02-run-local",
            step_id="lab-02-run-local-venv",
            order="pre",
        )

tests/test_items.py:48: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

client = <starlette.testclient.TestClient object at 0x7f7bbb593d90>, task_id = 'lab-02-run-local', step_id = 'lab-02-run-local-venv', order = 'pre'

    def get_task_by_id(client: TestClient, task_id: str, step_id: str, order: str):
        response = client.get(f"/items/item/{task_id}?order={order}")
    
>       assert response.status_code == 200
E       assert 404 == 200
E        +  where 404 = <Response [404 Not Found]>.status_code

tests/test_items.py:35: AssertionError
=============================================================== short test summary info ===============================================================
FAILED tests/test_items.py::test_get_item_1 - assert 404 == 200
============================================================= 1 failed, 3 passed in 0.34s =============================================================

--- a/src/app/services/item_service.py
+++ b/src/app/services/item_service.py
@@ -230,11 +230,14 @@ def get_item_by_id_dfs_iterative(
                         counter += 1
                         if lab.id == item_id:
                             return FoundItem(task, counter)
+                        if task.id == item_id:
+                            return FoundItem(task, counter)
 
                         for step in task.steps:
                             counter += 1
                             if step.id == item_id:
                                 return FoundItem(step, counter)
+
         case PostOrder():
             # TODO implement
             pass

